dbms/src/DataStreams/WindowTransformAction.cpp

```diff
@@ -239,8 +239,11 @@ WindowTransformAction::WindowTransformAction(
     }
 
     initialWorkspaces();
-
     initialPartitionAndOrderColumnIndices();
+
+    has_shortcut = window_description_.frame.begin_type == WindowFrame::BoundaryType::Unbounded
+        && window_description_.frame.end_type == WindowFrame::BoundaryType::Unbounded
+        && !aggregation_workspaces.empty();
```

Comment:
and `window_workspaces` should be empty?

Reply:
> and `window_workspaces` should be empty?

yes

Reply:
I mean you should also add this check

Reply:
> I mean you should also add this check

done